test/dart_eval_test.dart

```diff
@@ -876,6 +876,301 @@ void main() {
+              const str = "D";
+              return str.padLeft(4, "y");
+            }
+          ''',
+        }
+      });
+      expect((exec.executeLib('package:example/main.dart', 'test') as $String).$value, 'yyyD');
+    });
+    test('String padRight method formats with default padding', () {
+      final exec = compiler.compileWriteAndLoad({
+        'example': {
+          'main.dart': '''
+            String test() {
+              const str = "D";
+              return str.padRight(4);
+            }
+          ''',
+        }
+      });
+      expect((exec.executeLib('package:example/main.dart', 'test') as $String).$value, 'D   ');
+    });
+    test('String padRight method formats with given padding', () {
+      final exec = compiler.compileWriteAndLoad({
+        'example': {
+          'main.dart': '''
+            String test() {
+              const str = "D";
+              return str.padRight(4, "y");
+            }
+          ''',
+        }
+      });
+      expect((exec.executeLib('package:example/main.dart', 'test') as $String).$value, 'Dyyy');
+    });
+    test('String replaceAll method replaces text', () {
+      final exec = compiler.compileWriteAndLoad({
+        'example': {
+          'main.dart': '''
+            String test() {
+              const cat = "Fluffy";
+              return cat.replaceAll("f", "z");
+            }
+          ''',
+        }
+      });
+      expect((exec.executeLib('package:example/main.dart', 'test') as $String).$value, 'Fluzzy');
+    });
+    test('String replaceFirst method replaces with default start index', () {
+      final exec = compiler.compileWriteAndLoad({
+        'example': {
+          'main.dart': '''
+            String test() {
+              const cat = "Fluffy";
+              return cat.replaceFirst('f', 'z');
+            }
+          ''',
+        }
+      });
+      expect((exec.executeLib('package:example/main.dart', 'test') as $String).$value, 'Fluzfy');
+    });
+    test('String replaceFirst method replaces with given start index', () {
+      final exec = compiler.compileWriteAndLoad({
+        'example': {
+          'main.dart': '''
+            String test() {
+              const cat = "Fluffy";
+              return cat.replaceFirst('f', 'z', 4);
+            }
+          ''',
+        }
+      });
+      expect((exec.executeLib('package:example/main.dart', 'test') as $String).$value, 'Flufzy');
+    });
+    test('String toString method returns same string', () {
+      final exec = compiler.compileWriteAndLoad({
+        'example': {
+          'main.dart': '''
+            String test() {
+              const cat = "Fluffy";
+              return cat.toString();
+            }
+          ''',
+        }
+      });
+      expect((exec.executeLib('package:example/main.dart', 'test') as $String).$value, 'Fluffy');
+    });
+
+// Doesn't work right now, seems to ignore that KnownMethodArg has nullable set to true and get this error:
```

Comment:
@ethanblake4 I'm not sure if I'm doing something wrong here or if it is a bug in the nullable parameter handling?